codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/generators/PythonServerUnionGenerator.kt

```diff
@@ -0,0 +1,203 @@
+ */
+class PythonServerUnionGenerator(
+    model: Model,
+    private val symbolProvider: SymbolProvider,
+    private val writer: RustWriter,
+    shape: UnionShape,
+    private val renderUnknownVariant: Boolean = true,
+) : UnionGenerator(model, symbolProvider, writer, shape, renderUnknownVariant) {
+    private val sortedMembers: List<MemberShape> = shape.allMembers.values.sortedBy { symbolProvider.toMemberName(it) }
+    private val unionSymbol = symbolProvider.toSymbol(shape)
+
+    private val pyo3 = PythonServerCargoDependency.PyO3.toType()
+
+    override fun render() {
+        super.render()
+        renderPyUnionStruct()
+        renderPyUnionImpl()
+        renderPyObjectConverters()
+    }
+
+    private fun renderPyUnionStruct() {
+        writer.rust("""##[pyo3::pyclass(name = "${unionSymbol.name}")]""")
+        val containerMeta = unionSymbol.expectRustMetadata()
+        containerMeta.render(writer)
+        writer.rust("struct PyUnionMarker${unionSymbol.name}(pub ${unionSymbol.name});")
+    }
+
+    private fun renderPyUnionImpl() {
+        Attribute(pyo3.resolve("pymethods")).render(writer)
+        writer.rustBlock("impl PyUnionMarker${unionSymbol.name}") {
+            sortedMembers.forEach { member ->
+                val funcNamePart = member.memberName.toSnakeCase()
+                val variantName = symbolProvider.toMemberName(member)
+
+                if (sortedMembers.size == 1) {
+                    Attribute.AllowIrrefutableLetPatterns.render(this)
+                }
+                renderNewVariant(writer, model, symbolProvider, member, variantName, funcNamePart, unionSymbol)
+                renderAsVariant(writer, model, symbolProvider, member, variantName, funcNamePart, unionSymbol)
+                rust("/// Returns true if this is a [`$variantName`](#T::$variantName).", unionSymbol)
+                rust("/// :rtype bool:")
+                rustBlock("pub fn is_$funcNamePart(&self) -> bool") {
+                    rust("self.0.is_$funcNamePart()")
+                }
+            }
+            if (renderUnknownVariant) {
+                rust("/// Returns true if the union instance is the `Unknown` variant.")
+                rust("/// :rtype bool:")
+                rustBlock("pub fn is_unknown(&self) -> bool") {
+                    rust("self.0.is_unknown()")
+                }
+            }
+        }
+    }
+
+    private fun renderPyObjectConverters() {
+        writer.rustBlockTemplate("impl #{pyo3}::IntoPy<#{pyo3}::PyObject> for ${unionSymbol.name}", "pyo3" to pyo3) {
+            rustBlockTemplate("fn into_py(self, py: #{pyo3}::Python<'_>) -> #{pyo3}::PyObject", "pyo3" to pyo3) {
+                rustBlock("match self") {
+                    sortedMembers.forEach { member ->
+                        val variantName = symbolProvider.toMemberName(member)
+                        rust("${unionSymbol.name}::$variantName(variant) => variant.into_py(py),")
+                    }
+                }
```

Comment:
I think this should be just:
```suggestion
                rust("PyUnionMarker${unionSymbol.name}(self).into_py(py)")
```

because we're returning directly variants here, and we don't get the `is_{variant}`, `as_{variant}` methods. We should just return the wrapper type

Reply:
Addressed with https://github.com/awslabs/smithy-rs/pull/2427/commits/1fc0a92837c14aebb39f9666004e207d0cac8a6b